sbi/diagnostics/lc2st.py

```diff
@@ -0,0 +1,730 @@
+        if return_probas:
+            return probas, scores
+        else:
+            return scores
+
+    def train_data(self) -> None:
+        """Train the classifiers on the observed data.
+        Saves the trained classifiers.
+        """
+        trained_clfs = self._train(self.P, self.Q, self.x_P, self.x_Q)
+        self.trained_clfs = trained_clfs
+
+    def scores_data(
+        self, P_eval: Tensor, x_eval: Tensor, return_probas: bool = False
+    ) -> Union[np.ndarray, Tuple[np.ndarray, np.ndarray]]:
+        """Compute the L-C2ST scores for the observed data.
+
+        Args:
+            P_eval: Samples from P conditioned on the observation `x_eval`,
+                of shape (sample_size, dim).
+            x_eval: Observation, of shape (,dim_x).
+            return_probas: Whether to return the predicted probabilities of being in P,
+                defaults to False.
+
+        Returns: one of
+            scores: L-C2ST scores at `x_eval`.
+            (probas, scores): Predicted probabilities and L-C2ST scores at `x_eval`.
+        """
+        assert (
+            self.trained_clfs is not None
+        ), "No trained classifiers found. Run `train_data` first."
+        return self._scores(
+            P_eval=P_eval,
+            x_eval=x_eval,
+            trained_clfs=self.trained_clfs,
+            return_probas=return_probas,
+        )
+
+    def statistic_data(
+        self,
+        P_eval: Tensor,
+        x_eval: Tensor,
+        return_probas: bool = False,
+    ) -> Union[float, Tuple[np.ndarray, float]]:
+        """Computes the L-C2ST statistic for the observed data.
+
+        Args:
+            P_eval: Samples from P conditioned on the observation `x_eval`,
+                of shape (sample_size, dim).
+            x_eval: Observation, of shape (, dim_x)
+            return_probas: Whether to return the predicted probabilities of being in P,
+                defaults to False.
+
+        Returns:
+            L-C2ST statistic at `x_eval`.
+        """
+        probas, scores = self.scores_data(
+            P_eval=P_eval, x_eval=x_eval, return_probas=True
+        )
+        if return_probas:
+            return probas, scores.mean()
+        else:
+            return scores.mean()
+
+    def p_value(
+        self,
+        P_eval: Tensor,
+        x_eval: Tensor,
+    ) -> float:
+        """Computes the p-value for L-C2ST.
+
+        Args:
+            P_eval: Samples from P conditioned on the observation `x_eval`,
+                of dhape (sample_size, dim).
+            x_eval: Observation, of shape (, dim_x).
+
+        Returns:
+            p-value for L-C2ST at `x_eval`.
+        """
+        _, stat_data = self.statistic_data(
+            P_eval=P_eval, x_eval=x_eval, return_probas=True
+        )
+        _, stats_null = self.statistics_null(
+            P_eval=P_eval, x_eval=x_eval, return_probas=True, verbosity=0
+        )
+        return (stat_data < stats_null).mean()
+
+    def reject(
```

Comment:
I suggest to make it very explicit: 
```suggestion
    def p_value_smaller_alpha(
```

then you can things like 
```python
if p_value_smaller_alpha:
    ...
```

Reply:
As you which. This is just classical hypothesis test vocabulary. I changed it for now, we can always revert.